### DISCUSSION TITLE
loki::Search and costing equivalent of pass-through filter

### DISCUSSION THREAD

**Discussion by mloskot** [valhalla/valhalla#3254] [Q&A] [answered]
### Valhalla 3.0.9

Up to Valhalla 3.0.9, the `loki::Search` accepted optional `DynamicCost`

https://github.com/valhalla/valhalla/blob/ea06e876a5c997b84119892ab90a825fc6c2c5df/valhalla/loki/search.h#L29-L31

If a user did `loki::Search(locations, graph_reader);` then edge and node filters were set to pass-through:

https://github.com/valhalla/valhalla/blob/ea06e876a5c997b84119892ab90a825fc6c2c5df/src/loki/search.cc#L222-L223

which were defined as follows:

https://github.com/valhalla/valhalla/blob/ea06e876a5c997b84119892ab90a825fc6c2c5df/src/loki/search.cc#L21-L27

### Valhalla 3.1.x

The `DynamicCost` is no longer optional:

https://github.com/valhalla/valhalla/blob/d2914a4527f2aeb7495dfdbacea60fdc0b92c6a0/valhalla/loki/search.h#L29-L31

as it is a hard requirement:

https://github.com/valhalla/valhalla/blob/d2914a4527f2aeb7495dfdbacea60fdc0b92c6a0/src/loki/search.cc#L789-L790

### Question 

**What is the costing equivalent of the pass-through filters?**

Is it the `Costing::none_` costing?

It looks like it could be as `NoCost::Allowed` is now equivalent of the previous `PassThroughEdgeFilter` displayed above:

https://github.com/valhalla/valhalla/blob/d2914a4527f2aeb7495dfdbacea60fdc0b92c6a0/src/sif/nocost.cc#L203-L205

**Comment by kevinkreiser** [comment, marked as answer]
That's right the 'none' costing is the replacement

**Reply by mloskot** [reply]
Great, thanks.
